File: mybatis/src/main/java/data/spring/mybatis/application/required/product/ProductRepository.java
package data.spring.mybatis.application.required.product;

import data.spring.mybatis.application.service.product.command.ProductSearchCommand;
import data.spring.mybatis.application.service.product.command.ProductUpdateCommand;
import data.spring.mybatis.domain.product.Product;

import java.util.List;
import java.util.Optional;

public interface ProductRepository {
    void save(Product productEntity);

    int saveAll(List<Product> productEntities);

    void update(ProductUpdateCommand updateCommand);

    Optional<Product> findById(Long productId);

    List<Product> findAll(ProductSearchCommand searchCommand);

    int deleteAll();
}

File: mybatis/src/main/kotlin/data/spring/mybatis/application/required/product/ProductRepository.kt
package data.spring.mybatis.application.required.product

import data.spring.mybatis.application.service.product.command.ProductSearchCommand
import data.spring.mybatis.application.service.product.command.ProductUpdateCommand
import data.spring.mybatis.domain.product.Product
import java.util.*

interface ProductRepository : ProductRepositoryEx {
    fun save(product: Product)

    fun saveAll(products: List<Product>): Int

    fun findById(productId: Long): Product?

    fun findWithCond(searchCommand: ProductSearchCommand): List<Product>

    fun deleteAll(): Int
}